src/ITensorNetworks/peps.jl

```diff
@@ -106,17 +106,167 @@ end
 
 ITensors.data(P::PEPS) = P.data
 
-split_network(P::PEPS) = PEPS(split_network(data(P)))
+split_network(P::PEPS, rotation=false) = PEPS(split_network(data(P); rotation=rotation))
 
 function ITensors.commoninds(p1::PEPS, p2::PEPS)
   return mapreduce(a -> commoninds(a...), vcat, zip(p1.data, p2.data))
 end
 
+"""Returns a tree structure for a line of tensors with projectors
+Parameters
+----------
+line_size: size of the line structure
+center_index: the center index of the line
+site_tensors: a function, site_tensors(i) returns a list of tensors at position i
+projectors: the projectors of the line structure
+Returns
+-------
+Two trees, one in front of the center_index and another one after center_index
+Example
+-------
+   |  |   |
+p1-p2 |  p3-p4
+ | |  |   | |
+ | |  |   | |
+s1-s2-s3-s4-s5
+here line_size=5, center_index=3, si represents the list of tensors returned by site_tensors(i),
+projectors are [p1, p2, p3, p4].
+Returns two trees: [[s1, p1], s2, p2] and [[s5, p4], s4, p3]
```

Comment:
So the trees this returns already encode the optimal contraction sequence?

Also, in general the structure would contain projector tensors on both sides of the `si` tensors, right?

Reply:
yes that's right, in general projectors on both sides will be included.

Yes I already encode the optimal contraction sequence. The reason is that I find for large systems, cotengra still cannot find optimal path for the 1D structure consisting of a line and accompanying projectors. So finally just did it myself.

Reply:
Interesting. I find it strange that it even has trouble with that simple network, I think the more brute force optimization code in this module: https://github.com/ITensor/ITensors.jl/tree/main/src/ContractionSequenceOptimization could determine those networks without too much effort.

Reply:
But if you already hand-coded it, that is even better. We can work on more automation later.